src/features/client/components/Header/index.jsx

```diff
@@ -54,44 +59,107 @@ function Header() {
           justifyContent="space-between"
           alignItems="center"
         >
-          <Flex flexWrap="wrap" gap="10px" alignItems="center">
+          <Flex flexWrap="nowrap" gap="10px" alignItems="center">
             <Link as={ReachLink} to="/">
               <Image boxSize="30px" src={logoUrl} />
             </Link>
-            <Button variant="ghost" display="flex" gap="10px">
-              <Image
-                boxSize="30px"
-                src={locationUrl}
-                padding="5px"
-                rounded="full"
-                bg="blue.400"
-              />
-              <Text fontWeight="light">
-                Delivery to
-                <br />
-                <Box as="span" color="blue.400" fontWeight="semibold">
-                  Vilnius
-                </Box>
-              </Text>
-              <Image boxSize="20px" src={arrowDownUrl} />
-            </Button>
-          </Flex>
-          <Box>
-            <ButtonGroup gap={{ base: '5px', sm: '20px' }}>
-              <Button onClick={onLoginModalOpen} variant="ghost">
-                Log In
-              </Button>
-              <Button
-                onClick={onSignupModalOpen}
-                colorScheme="blue"
-                color="white"
-              >
-                Sign up
+            {smallerScreen ? (
+              <></>
+            ) : (
+              <Button variant="ghost" display="flex" gap="10px">
+                <Image
+                  boxSize="30px"
+                  src={locationUrl}
+                  padding="5px"
+                  rounded="full"
+                  bg="blue.400"
+                />
+                <Text fontWeight="light">
+                  Delivery to
+                  <br />
+                  <Box as="span" color="blue.400" fontWeight="semibold">
+                    Vilnius
+                  </Box>
+                </Text>
+                <Image boxSize="20px" src={arrowDownUrl} />
               </Button>
-            </ButtonGroup>
-            <BasketModal basket={basket} />
-          </Box>
+            )}
+          </Flex>
+          <Input
+            color="white"
+            bg="lightgray"
+            placeholder={
+              smallerScreen
+                ? 'Search for meal...'
+                : 'Search for cuisine, meal, restaurant...'
+            }
+            width={smallerScreen ? '40%' : '300px'}
+            borderRadius="50px"
+          />
+          {smallerScreen ? (
+            <Menu>
+              <MenuButton as={Button} rightIcon={<ChevronDownIcon />}>
+                User
+              </MenuButton>
+              <MenuList>
+                <MenuItem>
```

Comment:
This makes the container a button and child button. Bad practice to have interactive element inside another interactive element, explore documentation, there is a better way to construct this part.